sklearn/metrics/_pairwise_distances_reduction/_dispatcher.py

```diff
@@ -421,6 +428,10 @@ def compute(
         for the concrete implementation are therefore freed when this classmethod
         returns.
         """
+        # TODO: to maintain RAII, look at the implementation of the compute method
+        # if metric == 'precomputed':
+        #   return PrecomputedDistanceMatrix.precomputed_distance()
```

Comment:
`DatasetsPair.get_for` must be adapted accordingly to return an instance of the new class `PrecomputedDistanceMatrix{32,64}` you have created when `precomputed_matrix != None`.

Then, the precomputed distance matrix simply need to be forward to the implementation `compute` method and be pass to the `DatasetsPair.get_for` class method (e.g. see [this line for `ArgKmin`](https://github.com/scikit-learn/scikit-learn/blob/a97d584196da14084b33b71d80712451e26a5b0a/sklearn/metrics/_pairwise_distances_reduction/_argkmin.pyx.tp#90). 

Similar changes must be made to the other class (e.g. `{ArgKmin,RadiusNeighbors}{ClassMode,}{32,64}`, excepted the ones for the Euclidean specialization (e.g. `Euclidean{ArgKmin,RadiusNeighbors}{32,64}`).